📖 (doc): Upgrade Migration and Documentation over Alpha Commands

camilamacedo86 opened the pull request:
This PR includes the following improvements to the documentation:

- **Update the Migrations section** to clarify the scaffold upgrade process and highlight available automation options.
- **Introduce a new "Alpha Commands" section** to replace the outdated "Project Upgrade Assistance" content. This section provides a structured entry point for experimental upgrade tooling.
- **Add a dedicated page for `kubebuilder alpha update`**, detailing how the command works, when to use it, and how to safely automate project upgrades using the 3-way merge strategy.

These changes aim to improve the developer experience and provide clearer guidance for upgrading Kubebuilder projects across versions.

To preview the changes in the docs see:

- https://deploy-preview-4923--kubebuilder.netlify.app/reference/alpha_commands (Alpha commands section)
- https://deploy-preview-4923--kubebuilder.netlify.app/migrations ( Migrations )

vitorfloriano commented on docs/book/src/reference/commands/alpha_generate.md:
```diff
@@ -0,0 +1,93 @@
+# Regenerate your project with (`alpha generate`)
```
It seems the hyperlink to the alpha generate section is not working properly.

camilamacedo86 replied:
It seems solved now, if not please let me know what link is missing or not working

vitorfloriano commented on docs/book/src/reference/commands/alpha_update.md:
```diff
@@ -0,0 +1,93 @@
+# Update Your Project with (`alpha update`)
```
It seems the hyperlink isn't working properly.

camilamacedo86 replied:
It seems solved now, if not please let me know what link is missing or not working

vitorfloriano replied:
This is how I see it in the preview: ![image](https://github.com/user-attachments/assets/1e1b1cb3-a589-498c-8b86-d27202219f6a)
Not sure if that's how it's supposed to look.

vitorfloriano commented on docs/book/src/reference/commands/alpha_update.md:
```diff
@@ -0,0 +1,93 @@
+
+The `kubebuilder alpha update` command helps you upgrade your project scaffold to a newer Kubebuilder version or plugin layout automatically.
+
+It uses a **3-way merge strategy** to update your project with less manual work:
```
```suggestion
It uses a **3-way merge strategy** to update your project with less manual work. To achieve that, the command creates the following branches:
```

vitorfloriano commented on docs/book/src/reference/commands/alpha_update.md:
```diff
@@ -0,0 +1,93 @@
+This command creates temporary Git branches and deletes files during the process.
+
+Make sure to commit your work before running it.
+Temporary branches starting with `tmp-kb-update-` will be created to manage the upgrade.
```
Perhaps we should not use "temporary branches", since they will remain after the operation.

camilamacedo86 replied:
Changed

vitorfloriano commented on docs/book/src/reference/commands/alpha_update.md:
```diff
@@ -0,0 +1,93 @@
+2. Creates `tmp-kb-update-ancestor` with a clean scaffold using that version
+3. Creates `tmp-kb-update-current` and restores your current code on top
+4. Creates `tmp-kb-update-upgrade` using the latest scaffold
+5. Merges everything into `tmp-kb-update-merge` for review and testing
```
Perhaps we shouldn't use "merges everything", since it merges only current state of the project (in the merge branch) and the upgrade branch to perform the three-way merge.

vitorfloriano commented on docs/book/src/reference/commands/alpha_update.md:
````diff
@@ -0,0 +1,93 @@
+If needed, set a specific version or branch:
+
+```sh
+kubebuilder alpha update \
+  --from-version=v4.5.2 \
+  --to-version=v4.6.0 \
+  --from-branch=main
````
Perhaps we could use a cleaner example:
```suggestion
kubebuilder alpha update --from-version v4.5.2 --to-version v4.6.0 --from-branch main
```

camilamacedo86 replied:
The \
is to make clear the command and allow people copy and paste and just run

vitorfloriano commented on docs/book/src/reference/commands/alpha_update.md:
```diff
@@ -0,0 +1,93 @@
+
+| Flag                | Description                                                                  |
+|---------------------|------------------------------------------------------------------------------|
+| `--from-version`    | Kubebuilder version your project was created with. If unset, uses the `PROJECT` file. |
+| `--to-version`      | Version to upgrade to. Defaults to the latest version.                       |
+| `--from-branch`     | Git branch that contains your current project code. Defaults to `main`.      |
+| `-h, --help`        | Show help for this command.                                                   |
+
+## Requirements
+
+- A valid [PROJECT][project-config] file at the root of your project
```
We could include a note that explicitly tells that the command uses the `cliVersion` field in the `PROJECT` file, and that it was introduced in `v4.6.0`, so the `--from-version` flag may be required for versions older than that.

camilamacedo86 replied:
Yep I think it is a good idea 👍

camilamacedo86 commented on docs/book/src/migrations.md:
```diff
@@ -1,8 +1,78 @@
+
+All inputs used by Kubebuilder are tracked in the `PROJECT` file.
+If you use the CLI to generate your scaffolds, this file will record the project's configuration and metadata.
+It plays a central role during automated migration processes by providing the required
```
Fixed

camilamacedo86 commented on docs/book/src/migrations.md:
```diff
@@ -1,8 +1,78 @@
+
+After using the CLI to create your project, you are free to customise how you see fit. Bear in mind, that it is not recommended to deviate from the proposed layout unless you know what you are doing.
+
+For example, you should refrain from moving the scaffolded files, doing so will make it difficult in upgrading your project in the future. You may also lose the ability to use some of the CLI features and helpers. For further information on the project layout, see the doc [What's in a basic project?][basic-project-doc]
```
fixed

camilamacedo86 commented:
Hi @vitorfloriano ,

Thank you so much for your review!
I’ve addressed all the feedback—could you please take another look when you have a moment?

Appreciate your time!

vitorfloriano commented:
/lgtm

camilamacedo86 commented:
moving forward after the review for we do a release
